torch/nn/modules/transformer.py

```diff
@@ -121,21 +122,21 @@ def forward(self, src, tgt, src_mask=None, tgt_mask=None,
                               memory_key_padding_mask=memory_key_padding_mask)
         return output
 
-    def generate_square_subsequent_mask(self, sz):
-        r"""Generate a square mask for the sequence. The masked positions are filled with float('-inf').
-            Unmasked positions are filled with float(0.0).
-        """
-        mask = (torch.triu(torch.ones(sz, sz)) == 1).transpose(0, 1)
-        mask = mask.float().masked_fill(mask == 0, float('-inf')).masked_fill(mask == 1, float(0.0))
-        return mask
-
     def _reset_parameters(self):
         r"""Initiate parameters in the transformer model."""
 
         for p in self.parameters():
             if p.dim() > 1:
                 xavier_uniform_(p)
 
+    def generate_square_subsequent_mask(self, sz):
```

Comment:
IMO, we should eventually retire this function since it's not relevant to any components in `nn.Transformer`. You could add a warning message for this func and we could retire it after two releases. @cpuhrsch

Reply:
Yes that sounds like a reasonable approach. However, this function is simple enough that I'm not sure it warrants being a separate entry in our nn.functional module given the sort of other functions it would neighbor (e.g. conv2d).

@rudvlf0413 is there an advantage to having this be its own functional or static method?

Reply:
@cpuhrsch We can use the function without creating the Transformer object if we separate the function from the Transformer module.
This is helpful when we custom Transformer module or order modules that require that function.

Reply:
@rudvlf0413 given the code is very short, would your usecase be served by copy-pasting the code?

Reply:
@cpuhrsch Yeah, Although the function is separated from the main module, the body of the function is the same. However, we will feel convenient if we can use the function directly.

Reply:
@cpuhrsch I feel it's really a simple functional and is not necessary to keep it around.  @rudvlf0413 could you give us some usage cases, which are not related to `nn.Transformer` modules. :)

Reply:
@cpuhrsch @zhangguanheng66 I think this function can be applied to other seq-to-seq models such as "Convolutional sequence to sequence learning (https://arxiv.org/abs/1705.03122)" that requires masking operations when generating target sequences in training session.

I found a similar function from FairSeq as below
https://github.com/pytorch/fairseq/blob/master/fairseq/models/transformer.py#L593

Although the transformer module only uses generate_square_subsequent_mask currently, I think some modules will need this function in the future. Thus, I think it is better to locate the function in nn.functional.